pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -4640,3 +4641,206 @@ func TestSimplifiedMonitoringSwitch(t *testing.T) {
 
 	assert.Nil(t, env.Console.PersistentVolumeClaims, "Should not have PVC!")
 }
+
+func TestSmartRouterDefaultConf(t *testing.T) {
+	name := "test"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamAuthoringHA,
+			Objects: api.KieAppObjects{
+				SmartRouter: createSmartRouterEmpty(),
+			},
+		},
+	}
+	env, _ := GetEnvironment(cr, test.MockService())
+	testDefaultSmartRouter(t, env.SmartRouter.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image, cr.Status.Applied.Version, cr.Status.Applied.Objects.SmartRouter.ImageContext)
+}
+
+func createSmartRouterEmpty() *api.SmartRouterObject {
+	smartRouter := api.SmartRouterObject{
+		KieAppObject:     api.KieAppObject{},
+		Protocol:         "",
+		UseExternalRoute: false,
+	}
+	return &smartRouter
+}
+
+func testDefaultSmartRouter(t *testing.T, image string, version string, context string) {
+	if context != "" {
+		assert.Equal(t, context+"/"+constants.RhpamPrefix+"-smartrouter"+constants.RhelVersion+":"+version, image)
+	} else {
+		assert.Equal(t, constants.RhpamPrefix+"-smartrouter"+constants.RhelVersion+":"+version, image)
+	}
+}
+
+func TestSmartRouterWithImageContext(t *testing.T) {
+	name := "test"
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamAuthoringHA,
+			Objects: api.KieAppObjects{
+				SmartRouter: createSmartRouter(),
+			},
+		},
+	}
+	env, _ := GetEnvironment(cr, test.MockService())
+	testDefaultSmartRouter(t, env.SmartRouter.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image, cr.Status.Applied.Version, cr.Status.Applied.Objects.SmartRouter.ImageContext)
+}
+
+func createSmartRouter() *api.SmartRouterObject {
+	smartRouter := api.SmartRouterObject{
+		KieAppObject: api.KieAppObject{
+			ImageContext: "rhpam-42",
+		},
+		Protocol:         "",
+		UseExternalRoute: false,
+	}
+	return &smartRouter
+}
+
+func TestConsoleDefault(t *testing.T) {
```

Comment:
might want to give these new default tests a slightly more descriptive name? `TestConsoleDefaultImage` maybe?

Reply:
ok